### DISCUSSION TITLE
Localization support

### DISCUSSION THREAD

**Discussion by tsl0922** [tsl0922/ImPlay#24] [Announcements]
Localization support was added to the latest main branch.

We support the following languages now:

- `en-US`: English (United States)
- `zh-Hans`: Chinese (Simplified)
- `it-IT`: Italian (Italy)

To add a new translation for your language, just create a new json file under [resources/romfs/lang](https://github.com/tsl0922/ImPlay/tree/main/resources/romfs/lang) from  [resources/romfs/lang/en-US.json](https://github.com/tsl0922/ImPlay/blob/main/resources/romfs/lang/en-US.json), and send a PR.

**Note:**

- use [IETF language tag](https://en.wikipedia.org/wiki/IETF_language_tag) for language code
- non-English language should change `"fallback": true` to `"fallback": false`

**Comment by natural-harmonia-gropius** [comment]
建议命名格式换 zh-Hans，因为存在 zh-hk-en 这样的情况，同时也可以避免未来或许会加入 zh-tw 引发争议

**Reply by tsl0922** [reply]
已修改，不过这里只是个代码，不会对显示产生实际影响。

**Comment by bovirus** [comment]
@tsl0922 

Please add in first post a note that for any additional language

    "fallback": true,

present in english language should be

    "fallback": false,

and for language code ID to check here

https://en.wikipedia.org/wiki/IETF_language_tag

**Reply by tsl0922** [reply]
done.

**Comment by bovirus** [comment]
@tsl0922 

Could you please consider if your source code can do it to modify the source code to use external language file as text file?
Thnaks.

**Reply by tsl0922** [reply]
will do it (#41).
